src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -275,28 +273,29 @@ def __del__(self) -> None:
         serialized_request = _common.serialize(self._request,
                                                self._request_serializer)
 
-        # NOTE(lidiz) asyncio.CancelledError is not a good transport for
-        # status, since the Task class do not cache the exact
-        # asyncio.CancelledError object. So, the solution is catching the error
-        # in Cython layer, then cancel the RPC and update the status, finally
-        # re-raise the CancelledError.
-        serialized_response = await self._channel.unary_unary(
-            self._method,
-            serialized_request,
-            self._deadline,
-            self._cancellation,
-            self._set_initial_metadata,
-            self._set_status,
-        )
+        # NOTE(lidiz) asyncio.CancelledError is not a good transport for status,
+        # because the asyncio.Task class do not cache the exception object.
+        # https://github.com/python/cpython/blob/edad4d89e357c92f70c0324b937845d652b20afd/Lib/asyncio/tasks.py#L785
+        try:
+            serialized_response = await self._cython_call.unary_unary(
+                serialized_request,
+                self._set_initial_metadata,
+                self._set_status,
+            )
+        except asyncio.CancelledError:
```

Comment:
question when are we making the assignation of a `self._code = grpc.StatusCode.CANCELLED` when a `asyncio.CancelledError` exception happens?